pkg/controllermanager/controller/project/project_control_reconcile.go

```diff
@@ -142,10 +142,13 @@ func (r *projectReconciler) reconcile(ctx context.Context, project *gardencorev1
 		}
 	}
 
-	// Update the project status to mark it as 'ready'.
+	// Update the project status to mark it as 'ready' and add activity timestamp if it is missing.
 	if err := updateStatus(ctx, gardenClient, project, func() {
 		project.Status.Phase = gardencorev1beta1.ProjectReady
 		project.Status.ObservedGeneration = generation
+		if project.Status.LastActivityTimestamp == nil {
+			project.Status.LastActivityTimestamp = &project.CreationTimestamp
+		}
```

Comment:
Can you elaborate why this is needed? Normally, [this check](https://github.com/gardener/gardener/blob/c71c50b53bb7de36a441d9d4df43cb1dd6ac89ae/pkg/controllermanager/controller/project/project_stale_control.go#L109) already considers newly created projects.

Reply:
The last activity timestamp may not be set on older projects so we need to initialise it first somewhere. Otherwise there would be panics in the activity reconciler. Thats why I have added [this](https://github.com/gardener/gardener/blob/3b0f8dc8eba36d78b63dba523f3f1b1065bb9808/pkg/controllermanager/controller/project/project_activity_control.go#L71) as well.

Reply:
> The last activity timestamp may not be set on older projects so we need to initialise it first somewhere.

Suggestion: Also set it `project_activity_control.go` because logically it belongs to that controller. If you want to take a shortcut there, then iterate over all shoots in the project namespace and take the newest one for the `LastActivityTimestamp`. This spares a lot of update calls for existing projects when the timestamp is set.

Reply:
A start of KCM will cause the `AddFunc` to be called for every shoot at a random order (at least I'm not aware of any specific order). Hence, if you have a project with shoots, but the project doesn't have a `LastActivityTimestamp`, then the reconciliation function will update the project status probably many times in a row until the newest `CreationTimestamp` is set.

Reply:
Ok now I see what you mean, thanks!